From b9f314c2e61df3836f50b0a2963b769d07b78b19 Mon Sep 17 00:00:00 2001
From: Fancy Fang <B47543@freescale.com>
Date: Thu, 19 Dec 2013 18:32:42 +0800
Subject: [PATCH 0677/1074] ENGR00292816 PXP: move two struct definitions to
 pxp_device.h

Move two struct definitions defined in pxp_device.c to pxp_device.h.
Now the pxp_device.h has been created for PXP device driver. So all
the type definition should stay in header file not c source file.

Signed-off-by: Fancy Fang <B47543@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/dma/pxp/pxp_device.c |   13 -------------
 include/linux/pxp_device.h   |   13 +++++++++++++
 2 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/drivers/dma/pxp/pxp_device.c b/drivers/dma/pxp/pxp_device.c
index 7d0ac40..2695e6e 100644
--- a/drivers/dma/pxp/pxp_device.c
+++ b/drivers/dma/pxp/pxp_device.c
@@ -36,19 +36,6 @@ static atomic_t open_count = ATOMIC_INIT(0);
 static DEFINE_SPINLOCK(pxp_mem_lock);
 static LIST_HEAD(head);
 
-/* To track the allocated memory buffer */
-struct memalloc_record {
-	struct list_head list;
-	struct pxp_mem_desc mem;
-};
-
-struct pxp_irq_info {
-	wait_queue_head_t waitq;
-	int irq_pending;
-	int hist_status;
-	spinlock_t lock;
-};
-
 static struct dma_chan *dma_chans[NR_PXP_VIRT_CHANNEL];
 static struct pxp_irq_info irq_info[NR_PXP_VIRT_CHANNEL];
 
diff --git a/include/linux/pxp_device.h b/include/linux/pxp_device.h
index 34d36bd..38120d8 100644
--- a/include/linux/pxp_device.h
+++ b/include/linux/pxp_device.h
@@ -21,4 +21,17 @@
 
 #include <uapi/linux/pxp_device.h>
 
+/* To track the allocated memory buffer */
+struct memalloc_record {
+	struct list_head list;
+	struct pxp_mem_desc mem;
+};
+
+struct pxp_irq_info {
+	wait_queue_head_t waitq;
+	int irq_pending;
+	int hist_status;
+	spinlock_t lock;
+};
+
 #endif
-- 
1.7.5.4

